case-lib/hijack.sh

```diff
@@ -185,6 +185,11 @@ function func_exit_handler()
         fi
     }
 
+    # Restore ALSA settings after execution if state file exists
+    if [ -f /var/tmp/"${SCRIPT_NAME##*/}".state ]; then
```

Comment:
I've been wanting to reduce SCRIPT_NAME usage in #546 for   a few years :-)

Reply:
```suggestion
    if [ -f /tmp/"${SCRIPT_NAME##*/}".state ]; then
```

While different distros do things differently, `/tmp` is in general on `tmpfs` ("RAM disk") whereas `/var/tmp` is persistent across reboots - which I'm pretty sure we do not WANT.

BTW did you consider using the process ID in the name? I mean `$$`

There is also https://en.wikipedia.org/wiki/TMPDIR, it's complicated :-(

Reply:
Here's another idea:
```
   SOF_TEST_ALSA_STATE=/tmp/"${SCRIPT_NAME##*/}".$$.state
```

... and then at restore time you can `test -n SOF_TEST_ALSA_STATE`? So the filename is defined in only one place.

Reply:
I chose `/var/tmp/` at first because I remembered that the DUT is rebooted in our CI during testing, but you're right - that reboot occurs before running the test case `.sh` file, so `/tmp` should be used instead.
I've applied your last suggestion and squashed the commits.
I've taken a look at the failed shellcheck and it all seemed like problems that already existed in the codebase, rather than something newly introduced.
PR checked in CI run No. 54699

Reply:
> but you're right - that reboot occurs before running the test case .sh file, so /tmp should be used instead.

There is no reboot test AFAIK right now:
- https://github.com/thesofproject/sof-test/issues/1038

PS: there are some suspend/resume tests but they are not very reliable (see https://github.com/thesofproject/sof-test/discussions/1017 and internal sof-framework/issues/408).